superset/assets/javascripts/explore/AdhocMetric.js

```diff
@@ -0,0 +1,37 @@
+export default class AdhocMetric {
+  constructor(adhocMetric) {
+    this.column = adhocMetric.column;
+    this.aggregate = adhocMetric.aggregate;
+    this.hasCustomLabel = !!(adhocMetric.hasCustomLabel && adhocMetric.label);
+    this.fromFormData = !!adhocMetric.optionName;
+
+    if (this.hasCustomLabel) {
```

Comment:
you could consolidate with 
`this.label = this.hasCustomLabel ? adhocMetric.label : this.getDefaultLabel();`